tsdoc/src/nodes/DocMemberSelector.ts

```diff
@@ -57,7 +57,7 @@ export interface IDocMemberSelectorParsedParameters extends IDocNodeParsedParame
 export class DocMemberSelector extends DocNode {
   private static readonly _likeIndexSelectorRegExp: RegExp = /^[0-9]/;
 
-  private static readonly _indexSelectorRegExp: RegExp = /^[1-9][0-9]*$/;
+  private static readonly _indexSelectorRegExp: RegExp = /^(0|[1-9][0-9]*)$/;
```

Comment:
@rugvip The numeric index selectors were designed to count starting from `1` not `0`.  API Extractor matches them to [ApiParameterListMixin.overloadIndex](https://github.com/microsoft/rushstack/blob/d0f8f10a9ce1ce4158ca2da5b79c54c71d028d89/libraries/api-extractor-model/src/mixins/ApiParameterListMixin.ts#L82):

```ts
 /**
   * When a function has multiple overloaded declarations, this zero-based integer index can be used to unqiuely
   * identify them.
   *
   * @remarks
   *
   * Consider this overloaded declaration:
   *
   * ```ts
   * export namespace Versioning {
   *   // TSDoc: Versioning.(addVersions:1)
   *   export function addVersions(x: number, y: number): number;
   *
   *   // TSDoc: Versioning.(addVersions:2)
   *   export function addVersions(x: string, y: string): string;
   *
   *   // (implementation)
   *   export function addVersions(x: number|string, y: number|string): number|string {
   *     // . . .
   *   }
   * }
   * ```
   *
   * In the above example, there are two overloaded declarations.  The overload using numbers will have
   * `overloadIndex = 1`.  The overload using strings will have `overloadIndex = 2`.  The third declaration that
   * accepts all possible inputs is considered part of the implementation, and is not processed by API Extractor.
   */
  readonly overloadIndex: number;
```

It seems that the TSDoc library comments are vague about this. We should improve it.

Reply:
Maybe it seems counterintuitive to count from `1` since computer arrays usually count from `0`. I think the rationale was that this numbering was meant to be used in the context of documentation.  For example, we normally count from `1` when referring to chapter numbers, page numbers, dictionary entries, bullet lists, etc.

Reply:
> 
> /**
> &nbsp;&nbsp;\* When a function has multiple overloaded declarations, this ***zero-based*** integer index can be used to unqiuely
> 

😂 Even this code comment is confused about it.

I double-checked the implementation, and it does indeed start from 1:

[api-extractor/src/collector/Collector.ts](https://github.com/microsoft/rushstack/blob/d0f8f10a9ce1ce4158ca2da5b79c54c71d028d89/apps/api-extractor/src/collector/Collector.ts#L407)
```ts
      // TSDoc index selectors are positive integers counting from 1
      let nextIndex: number = 1;
```